### PR TITLE
Create partner activity connect #222

### LINKED ISSUE #222: Be able to create a partner activity
2 modes:
* https://projects.invisionapp.com/share/EDBIM0Y8M#/screens/232533037
* https://projects.invisionapp.com/share/EDBIM0Y8M#/screens/232533038

When choosing add from cluster activity this new activity should be linked then to the parent cluster activity and it should get the same set of reportables / indicators as the parent activity does. Even those indicators should be linked to parent indicators that are associated with the cluster activity.

If they create a custom activity then its not related to cluster.

**technical notes**
Please see data model for the parent / child relationships between activity and indicators.

### PR DESCRIPTION
**Pull request by rkrupinski**
### This PR addresses resolve #222

### CODE REVIEW

File: polymer/src/elements/cluster-reporting/activity-modal.html [hunk 1]
```diff
@@ -0,0 +1,678 @@
...
+        },
+
+        clusters: {
+          type: Array,
+          value: [],
+        },
+
+        statuses: {
+          type: Array,
+          value: [
```
Comment (n1207n) [inline polymer/src/elements/cluster-reporting/activity-modal.html]: Should we put these status choices to some constant JS file? I doubt we're gonna add or remove some of these choices in the future, but at least we'll have a centralized place for all choice value lists and constants.

If it's too overkill, then don't worry much ;)

**Comment by rkrupinski** [comment]
@n1207n We spent some time w/ @lkolacz trying to resolve conflicts but with no result. Could you please merge `develop` to `create-partner-activity-#222`. Since both django & polymer codebases seem to have been reviewed now, feel free to merge the PR.

File: polymer/src/endpoints.html [hunk 2]
```diff
@@ -121,6 +125,9 @@
         return this._buildUrl('/indicator/cluster-indicator/');
       },
 
+      partnerActivity: function (responsePlanId, mode) {
+        return this._buildUrl('/partner/' + responsePlanId + '/create-partner-activity/' + mode);
+
```
Comment (ErinOC) [inline polymer/src/endpoints.html]: Missing the closing parenthesis.

File: polymer/src/elements/cluster-reporting/activity-modal.html [hunk 3]
```diff
@@ -0,0 +1,678 @@
...
+        projectsParams: {
+          type: Object,
+          value: {
+            page_size: 99999,
+          },
+        },
+
+        projectsUrl: {
+          type: String,
+          value: App.Endpoints.plannedActions(),
```
Comment (ErinOC) [inline polymer/src/elements/cluster-reporting/activity-modal.html]: After develop was merged into this branch, passing the response plan ID is required to create this URL.

File: polymer/src/elements/cluster-reporting/activity-modal.html [hunk 4]
```diff
@@ -0,0 +1,678 @@
...
+        },
+
+        projectsParams: {
+          type: Object,
+          value: {
+            page_size: 99999,
+          },
+        },
+
+        projectsUrl: {
```
Comment (ErinOC) [inline polymer/src/elements/cluster-reporting/activity-modal.html]: I'm getting this error when trying to save: {"non_field_errors":["PartnerProject does not belong to Partner 36."]}

The current partner ID should be passed to as a query string to get only the projects associated with this partner: partner-project-list/?partner=36

File: polymer/src/elements/cluster-reporting/activity-modal.html [hunk 5]
```diff
@@ -0,0 +1,678 @@
...
+        objectivesParams: {
+          type: Object,
+          value: {
+            page_size: 99999,
+          },
+        },
+
+        objectivesUrl: {
+          type: String,
+          value: App.Endpoints.responseParametersClusterObjectives(),
```
Comment (ErinOC) [inline polymer/src/elements/cluster-reporting/activity-modal.html]: Needs a response plan ID after merging in develop.

File: django_api/django_api/apps/partner/serializer.py [hunk 6]
```diff
@@ -152,6 +159,82 @@ def get_links(self, obj):
...
+        try:
+            data['partner'] = Partner.objects.get(id=data['partner'])
+        except Partner.DoesNotExist as e:
+            raise serializers.ValidationError(
+                'Partner ID {} does not exist.'.format(data['partner']))
+
+        try:
+            data['project'] = PartnerProject.objects.get(id=data['project'])
+
+            if data['project'].partner.id != self.initial_data['partner']:
```
Comment (ErinOC) [inline django_api/django_api/apps/partner/serializer.py]: I'm getting an error here when I try to save a custom activity using a project that I created: 
'NoneType' object has no attribute 'id'
In the project creation modal, the frontend does not pass back a partner ID (it's not in mockups and the API doesn't require it - I'd assume it should be assigned by the backend when creating a project).

File: polymer/src/endpoints.html (hunk 2 above)
Comment (rkrupinski) [inline polymer/src/endpoints.html]: Yep, had it locally; forgot to commit 🐸

File: polymer/src/elements/cluster-reporting/activity-modal.html (hunk 4 above)
Comment (rkrupinski) [inline polymer/src/elements/cluster-reporting/activity-modal.html]: Done

File: django_api/django_api/apps/partner/serializer.py (hunk 6 above)
Comment (rkrupinski) [inline django_api/django_api/apps/partner/serializer.py]: Managed to add both cluster & custom activity. If that's still an issue, let's file a bug for it.

File: polymer/src/elements/cluster-reporting/activity-modal.html (hunk 3 above)
Comment (rkrupinski) [inline polymer/src/elements/cluster-reporting/activity-modal.html]: Done

File: polymer/src/elements/cluster-reporting/activity-modal.html (hunk 5 above)
Comment (rkrupinski) [inline polymer/src/elements/cluster-reporting/activity-modal.html]: Done

**Comment by rkrupinski** [comment]
```
.........................................................................................
----------------------------------------------------------------------
Ran 89 tests in 1028.678s
```

Merging.

**Pull request merged**